lib/src/facts/linux/fips_resolver.cc

```diff
@@ -0,0 +1,35 @@
+#include <internal/facts/linux/fips_resolver.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+
+using namespace std;
+
+using boost::lexical_cast;
+using boost::bad_lexical_cast;
+
+namespace lth_file = leatherman::file_util;
+
+namespace facter { namespace facts { namespace linux {
+
+    fips_resolver::data fips_resolver::collect_data(collection& facts)
+    {
+        data result;
+
+        // Below file might not exist in which case we need to ensure to not report
+        // anything incorrectly by initializing the result to a safe value
+        result.is_fips_mode_enabled = false;
+
+        lth_file::each_line("/proc/sys/crypto/fips_enabled", [&](string& line) {
```

Comment:
You could simplify the entire code of this routine to:

result.is_fips_mode_enabled = boost::trim(lth_file::read("/proc/sys/crypto/fips_enabled")) != "0"

Note that if the file does not exist, lth_file::read would return the empty string (per https://github.com/puppetlabs/leatherman/blob/master/file_util/src/file.cc#L37) so your original default would still work.

Reply:
Agreeably there was a bit of redundancy particularly after I added the initialization but forgot to removing doing the same down below if fips_enabled read "0"

Reply:
Actually the above generates compile error perhaps becoz trim expects a reference string type as against a string object returned by read. It could be massaged but am planning to use the earlier check. Will remove the redundant assignment though in case fips is not enabled.